packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -14,11 +14,24 @@ let buildWrapper = template(`
   })
 `);
 
+let arrowBuildWrapper =  template(`
+  (() => {
+    var ref = FUNCTION;
+    return (PARAMS) => ref.apply(this, arguments);
```

Comment:
You can currently make an `exec.js` like https://github.com/babel/babel/blob/master/packages/babel-plugin-check-es2015-constants/test/fixtures/general/block-scoped/exec.js or https://github.com/frantic/babel/blob/hoist-file-name/packages/babel-plugin-transform-react-jsx-source/test/fixtures/react-source/basic-sample/exec.js, etc